src/test/java/fi/hsl/transitdata/cancellation/util/BulletinUtilsTest.java

```diff
@@ -0,0 +1,117 @@
+package fi.hsl.transitdata.cancellation.util;
+
+import fi.hsl.common.transitdata.proto.InternalMessages;
+import fi.hsl.transitdata.cancellation.domain.CancellationData;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+
+public class BulletinUtilsTest {
+    @Test
+    public void filterMassCancellationsFromEmptyBulletinsList() {
+        List<InternalMessages.Bulletin> inputBulletins = new ArrayList<>();
+        List<InternalMessages.Bulletin> outputBulletins = BulletinUtils.filterMassCancellationsFromBulletins(inputBulletins);
+        
+        assertTrue(outputBulletins.isEmpty());
+    }
+    
+    @Test
+    public void filterMassCancellations() {
+        InternalMessages.Bulletin bulletinMassCancellation1 = createBulletin(
+                InternalMessages.Bulletin.Impact.CANCELLED,
+                InternalMessages.Bulletin.Priority.WARNING,
+                "HSL:1111", "HSL:2222");
+        
+        InternalMessages.Bulletin bulletinOther1 = createBulletin(
+                InternalMessages.Bulletin.Impact.IRREGULAR_DEPARTURES,
+                InternalMessages.Bulletin.Priority.WARNING,
+                "HSL:3333", "HSL:4444");
+        
+        InternalMessages.Bulletin bulletinMassCancellation2 = createBulletin(
+                InternalMessages.Bulletin.Impact.CANCELLED,
+                InternalMessages.Bulletin.Priority.WARNING,
+                "HSL:5555", "HSL:5555");
+        
+        InternalMessages.Bulletin bulletinOther2 = createBulletin(
+                InternalMessages.Bulletin.Impact.RETURNING_TO_NORMAL,
+                InternalMessages.Bulletin.Priority.INFO,
+                "HSL:6666", "HSL:7777");
+        
+        InternalMessages.Bulletin bulletinOther3 = createBulletin(
+                InternalMessages.Bulletin.Impact.DISRUPTION_ROUTE,
+                InternalMessages.Bulletin.Priority.SEVERE,
+                "HSL:8888", "HSL:9999");
+        
+        List<InternalMessages.Bulletin> inputBulletins = new ArrayList<>();
+        inputBulletins.add(bulletinMassCancellation1);
+        inputBulletins.add(bulletinOther1);
+        inputBulletins.add(bulletinMassCancellation2);
+        inputBulletins.add(bulletinOther2);
+        inputBulletins.add(bulletinOther3);
```

Comment:
split this to own method `initializeTestBulletin` since it's not really relevant to the test itself. Makes reading test easier